hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -581,6 +580,15 @@ protected void runTableServicesInline(HoodieTable table, HoodieCommitMetadata me
     }
   }
 
+  private void inlineCompaction(HoodieTable table, Option<Map<String, String>> extraMetadata) {
```

Comment:
It's a good suggestion, Can we refactor this later？

Reply:
We just needto co ordinate with RDD APl users like uber, Since it's a breaking change. We can file a code cleanup follow-up JIRA, Orthogonal to this PR?

Reply:
Reading this again, a separate client impl is probably cleaner even for this PR. These if checks will grow harder to maintain over time.

Reply:
I have built a Jira to trace, we can do a refactoring of this later, Jira: https://issues.apache.org/jira/browse/HUDI-4407, WDYT?